### ISSUE TITLE
vsts-vscode no longer working with VS2017?

### ISSUE THREAD

**Issue by bms-lettschmidt** [Microsoft/vsts-vscode#211]
When using the VS2017 tf.exe we are no longer able to sync with our VSTS.
Error message is 
_[Sync] It appears you are using a Server workspace. Currently, TFVC support is limited to Local workspaces._  
but the workspace used is a local one. The colleagues using VS2015 tf.exe do not have any problems using the same project and configuration settings.

**Comment by jeffyoung** [comment]
Hi @bms-lettschmidt.  

Can you turn on [logging](https://github.com/Microsoft/vsts-vscode#logging), repro the issue and provide the log here?  You should get that error if we receive a message indicating that you don't have access to a particular folder.

Also, can you double-check the location of your workspace by following [these instructions](https://github.com/Microsoft/vsts-vscode/blob/master/TFVC_README.md#what-is-the-difference-between-a-local-and-server-workspace-how-can-i-tell-which-one-im-working-with)?  In the meantime, I'll try to reproduce it on my end.

**Comment by bms-lettschmidt** [comment]
Hi, sorry for the late reply. 
You are right, the error message in the log is 
TFVC errors: TF30063: You are not authorized to access bms-software.visualstudio.com\bms-software.

But I definitely have access to it (I am the owner of that project) and it si working fine from within VS2017.
I also double-checked the workspace settings, configuration looks fine.

On Monday I will compare all settings in VS Code & VS again with my colleagues using VS2015, maybe we will find some additional configuration differences than the VS version.

Here is the log, please let me know if you need additional information.

2017-05-07T05:36:56.290Z - info:   [03352] Polling for latest current build status...
2017-05-07T05:36:56.290Z - info:   [03352] Getting current build from badge...
2017-05-07T05:36:56.290Z - info:   [03352] Polling for the pinned work itemquery
2017-05-07T05:36:56.290Z - info:   [03352] Running pinned work item query to get count (Project X)...
2017-05-07T05:36:56.568Z - info:   [03352] No builds were found for team Project X, repo id 00000000-0000-0000-0000-000000000000, + branch UNKNOWN
2017-05-07T05:37:05.501Z - debug:  [03352] TFVC Repository.Sync
2017-05-07T05:37:05.501Z - debug:  [03352] TFVC: tf get -noprompt ******** -nosummary x:\bms-software\Project X -recursive
2017-05-07T05:37:05.505Z - debug:  [03352] TFVC: spawned new process (duration: 4ms)
2017-05-07T05:37:06.112Z - debug:  [03352] TFVC: get exit code: 100 (duration: 611ms)
2017-05-07T05:37:06.112Z - debug:  [03352] TFVC errors: TF30063: You are not authorized to access bms-software.visualstudio.com\bms-software.

**Comment by jeffyoung** [comment]
OK, so this seems to be the "server" workspace issue. That is, when you run `tf workfold` from the command line in that particular folder on disk, you'll get a screenshot similar to [this](https://github.com/Microsoft/vsts-vscode/raw/master/assets/tf-workspace-dialog.png).  If in doubt, can you provide a screenshot of the output of the `tf workfold` command?

Note that the `Location` field can either be `Local` or `Server`.  A server-based workspace results in the TF30063 error that you're seeing (and is issue #176). If you can use 'Local' workspaces ([here is documentation](https://www.visualstudio.com/en-us/docs/tfvc/decide-between-using-local-server-workspace) to help you make that determination), the extension will work properly.

**Comment by bms-lettschmidt** [comment]
The workspace is configured as a local one: 
![workspace_settings](https://cloud.githubusercontent.com/assets/28397789/25938680/5bd9716e-3630-11e7-9dd3-1ec39dcfcd25.png)

By the way, using the Visual Studio 2017 "tf workfold" command does only get me a text output:
```
Workspace : NB-LET_Project X (Lettschmidt, Torben)
Collection: https://bms-software.visualstudio.com/
 $/Project X: X:\bms-software\Project X
```

The UI I get by using the "tf workspace" command.

**Comment by bms-lettschmidt** [comment]
Hi again, no idea what changed - but since this afternoon it is working fine for my colleagues and me. I did not change any settings, but I do no longer get the error message.
Thanks for your support anyway!

**Comment by jeffyoung** [comment]
Awesome.  Thanks great to hear @bms-lettschmidt!

**Comment by FokkoVeegens** [comment]
Hi, I had the same issue. I was very sure I created a local workspace, I double-checked it. Happened with VS Code version 1.14.1, extension version 1.121.1. What I did to resolve the problem is: start Visual Studio 2017, change the workspace there to server workspace, let VS do its work, change it back to local workspace and again, let VS do its work. After that I was able to use VS code in combination with TFVC. Hope this helps others as well.

**Comment by dennispg** [comment]
I am having this issue and switching location to server and back to local does not work for me. I've tried a few times. I tried from VS2017 and from using `tf workspace`

Using VS Code version 1.14.2, extension version 1.121.1

Please reopen this!

**Comment by renzska** [comment]
I'm hitting this same issue, but I believe this Pull Request will resolve the issue once it is merged and released: https://github.com/Microsoft/vsts-vscode/pull/291

**Comment by jeffyoung** [comment]
I just released v1.122.0 that has the changes from #291 in it (which addresses the issue when using TF.exe and the client is on a different domain then the TFS server).  It's my hope this issue will be addressed by that change.

**Comment by dennispg** [comment]
v1.122.0 seems to have resolved the issue for me, however I had to use the form doman\username vs username@domain.com

**Comment by SRJames** [comment]
Hello. I've been getting this same issue for the past few days.
I have version 1.122.0 of the extension.
Version 1.15.1 of VS Code
Version 15.3.2 of Visual Studio Enterprise 2017

I am using .visualstudio.com with TFVC.
I have tried the "change to Server workspace and change back" trick - that hasn't worked.
I have reinstalled VStudio 2017 and the Extension. No change.
I have used both the PAT code authentication and the "new experience" method. No change.
With the logging set to debug I get the following
[VSCodeError.txt](https://github.com/Microsoft/vsts-vscode/files/1263355/VSCodeError.txt)

So I'm not sure if it is permissions or local v server workspace.
On the permissions front I certainly do have full access as I own all of this. 
BUT I am also experiencing this issue with VS 2017 which may just be a coincidence.
[https://developercommunity.visualstudio.com/content/problem/93366/tf30063-you-are-not-authorized-to-access-visualstu.html?childToView=104220#comment-104220](url)

Your help would be very much appreciated.

**Comment by jsvb1977** [comment]
I'm not sure why this is closed. Myself and others clearly also have this problem. Should I create a new issue?

**Comment by christopherdengler** [comment]
I am getting this issue too.  This is ridiculous.  Adding myself here in hopes that there's a fix to this.

**Comment by SideCoding** [comment]
Same issues here.

**Comment by philcatterall** [comment]
Same here

**Comment by MarkusWild** [comment]
I also have Problems connecting to a TFVC Project. The TFS Connection on a GIT Project on the same TFS Server works.
[team-extension.log](https://github.com/Microsoft/vsts-vscode/files/1456939/team-extension.log)
![icons](https://user-images.githubusercontent.com/11989851/32596684-6b4268b6-c534-11e7-9cb1-5a2e4f2a5fca.png)

**Comment by rafaelzuma** [comment]
I have tried all you had suggested here and the error continues!
Does anybody know other option to solution this?

**Comment by hmoratopcs** [comment]
In my case, the error appeared after I signed in with only my user name. I solved it signin in (F1 --> Team: Signin) with domain\username.

**Comment by rafaelzuma** [comment]
@hmoratopcs 
How do you sign with "domain\user" if "Team:Signin" requests our PAT (Personal Access Tokens)?
There are two option in "Team:Signin": current experience and new experience. Both shows the same error.

**Comment by hmoratopcs** [comment]
I was getting the error with an "on premises" Team Foundation Sever. I haven't tried with Team Services, sorry.

**Comment by rafaelzuma** [comment]
No problem, thanks for your help!
I am really heaving headache with VS Code conecting TFCV.
I don't know what I can do more...

**Comment by jhonalino** [comment]
having the same problem :( I have deadline tomorrow

**Comment by mccw** [comment]
I was having the same issue. I tried everything in this discussion and nothing worked. What finally worked for me was hitting F1, typing Team:Signout then hitting F1 again and then Team:Signin. I then supplied my PAT and currently appear to be talking to VSTS correctly.

**Comment by nmiddendorff** [comment]
This issue is related to Visual Studio 2017. I installed Visual Studio 2015 only to point to it's tf.exe file. 

VSTS worked after pointing the "tfvc.location" to "C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\Common7\\IDE\\tf.exe"

This solution may be overkill but fixes the issue until tf.exe is fixed in VS 2017.

**Comment by zatchgordon** [comment]
Although this issue is "closed" I've been trying to solve the same problem today.
I found my solution in this thread [here](https://github.com/Microsoft/vsts-vscode/issues/325#issuecomment-357871328).

I hope this helps
